pkg/controllers/provisioning/binpacking/packable.go

```diff
@@ -195,26 +193,13 @@ func (p *Packable) validateOperatingSystems(constraints *v1alpha5.Constraints) e
 	return nil
 }
 
-func (p *Packable) validateZones(constraints *v1alpha5.Constraints) error {
-	zones := sets.String{}
+func (p *Packable) validateOfferings(constraints *v1alpha5.Constraints) error {
 	for _, offering := range p.Offerings() {
-		zones.Insert(offering.Zone)
-	}
-	if constraints.Requirements.Zones().Intersection(zones).Len() == 0 {
-		return fmt.Errorf("zones %v are not in %v", zones, constraints.Requirements.Zones().List())
-	}
-	return nil
-}
-
-func (p *Packable) validateCapacityTypes(constraints *v1alpha5.Constraints) error {
-	capacityTypes := sets.String{}
-	for _, offering := range p.Offerings() {
-		capacityTypes.Insert(offering.CapacityType)
-	}
-	if constraints.Requirements.CapacityTypes().Intersection(capacityTypes).Len() == 0 {
-		return fmt.Errorf("capacity types %v are not in %v", capacityTypes, constraints.Requirements.CapacityTypes().List())
+		if constraints.Requirements.CapacityTypes().Has(offering.CapacityType) && constraints.Requirements.Zones().Has(offering.Zone) {
+			return nil
+		}
 	}
-	return nil
+	return fmt.Errorf("offerings %v are not available for capacity types %v and zones %v", p.Offerings(), constraints.Requirements.CapacityTypes(), constraints.Requirements.Zones())
```

Comment:
Can you show me an example of this string? Seems like it could be really verbose, especially printing offerings.

Reply:
```
offerings [{spot test-zone-1a} {on-demand test-zone-1b} {spot test-zone-1b}] are not available for capacity types [on-demand] and zones [test-zone-1a]
```

Reply:
Nice!